rmw_fastrtps_shared_cpp/src/listener_thread.cpp

```diff
@@ -154,26 +158,29 @@ node_listener(rmw_context_t * context)
     }
     if (subscriptions_buffer[0]) {
       rmw_dds_common::msg::ParticipantEntitiesInfo msg;
-      bool taken;
-      if (RMW_RET_OK != rmw_fastrtps_shared_cpp::__rmw_take(
-          context->implementation_identifier,
-          common_context->sub,
-          static_cast<void *>(&msg),
-          &taken,
-          nullptr))
-      {
-        TERMINATE_THREAD("__rmw_take failed");
-      }
-      if (taken) {
-        if (std::memcmp(
-            reinterpret_cast<char *>(common_context->gid.data),
-            reinterpret_cast<char *>(&msg.gid.data),
-            RMW_GID_STORAGE_SIZE) == 0)
+      bool taken = true;
```

Comment:
These changes are correct, but I would like to propose an additional change.

Avoid creating and destroying the waitset inside the main `while (common_context->thread_is_running.load())` loop

Reply:
Done in 6de684e4